Added reference to Attribute parameter types (#23831)
dotnet/docs: docs/csharp/programming-guide/concepts/attributes/index.md

The first parameter, the DLL name, is positional and always comes first; the others are named. In this case, both named parameters default to false, so they can be omitted. Positional parameters correspond to the parameters of the attribute constructor. Named or optional parameters correspond to either properties or fields of the attribute. Refer to the individual attribute's documentation for information on default parameter values.

> For more information on allowed parameter types refer to [Attributes](../../../../../_csharplang/spec/attributes.md#attribute-parameter-types) page in [C# Language Specification](../../../../../_csharplang/spec/introduction.md)

Reviewer: ```suggestion
For more information on allowed parameter types, see the [Attributes](~/_csharplang/spec/attributes.md#attribute-parameter-types) section of the [C# language specification](~/_csharplang/spec/introduction.md)
```
